
version: "3.7"
services:
  big-bear-planka:
    image: ghcr.io/plankanban/planka:1.26.3
    restart: unless-stopped
    volumes:
      - ${APP_DATA_DIR}/user-avatars:/app/public/user-avatars
      - ${APP_DATA_DIR}/project-background-images:/app/public/project-background-images
      - ${APP_DATA_DIR}/attachments:/app/private/attachments
    environment:
      - BASE_URL=http://[YOUR_CASAOS_LAN_IP]:3000
      - DATABASE_URL=postgresql://postgres@big-bear-planka-postgres/planka
      - SECRET_KEY=notsecretkey
      - DEFAULT_ADMIN_EMAIL=demo@demo.demo
      - DEFAULT_ADMIN_PASSWORD=demo
      - DEFAULT_ADMIN_NAME=Demo Demo
      - DEFAULT_ADMIN_USERNAME=demo
    networks:
      - big-bear-planka-network
    depends_on:
      big-bear-planka-postgres:
        condition: service_healthy
  big-bear-planka-postgres:
    image: postgres:14-alpine
    restart: on-failure
    volumes:
      - ${APP_DATA_DIR}/postgresql:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=planka
      - POSTGRES_HOST_AUTH_METHOD=trust
    networks:
      - big-bear-planka-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres -d planka"]
      interval: 10s
      timeout: 5s
      retries: 5
  app_proxy:
    environment:
      APP_HOST: big-bear-umbrel-planka_big-bear-planka_1
      APP_PORT: "1337"
networks:
  big-bear-planka-network:
    driver: bridge
